feat(errors): new request errors page for mobile monitoring (#18692)
newrelic/docs-website: src/content/docs/errors-inbox/group-request-errors.mdx

## View the request errors page [#request-errors-page]

To view the request errors page, go to **<DNT>[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Mobile > (select an app) > Request errors</DNT>**.

<img
    title="Request errors overview"
    style={{align: "left" }}
    alt="Screenshot of the request errors overview"
    src="/images/errors-inbox_screenshot-crop_request-errors-overview.webp"
/>

## Recommended workflow [#suggested-workflow]

We recommend the following approach to investigate HTTP request errors and network failures, allowing you to gain a complete understanding of the error, identify contributing factors, and implement effective debugging strategies:

1. **Drill down into a single error**: To view details about a request error or network failure, click on the row from the error group table. Here you can view the request information, request attributes, and response body as well as get more information about that error.

2. **Query and share error data**: To explore the data behind any of the charts or lists on the HTTP errors page:

>         * On any chart, click **… > <DNT>View query</DNT>**. 

Reviewer: ```suggestion
        * On any chart, click the **…** menu, and then click **<DNT>View query</DNT>**. 
```

        * From the query builder, you can add the error data to a dashboard and share it via a permalink.

	To dig deeper into the error data, query your data for the following events and attributes:

        * `MobileRequestError` events and attributes
        * `MobileRequest` events and attributes

3. **Change how the page groups and sorts errors and network failures**: Make selections using both the attributes and filter bars at the top of the page. By default, the errors are grouped by request domain and request path.

4. **Filter for specific errors and network failures**: Select an error or failure using multiple filters from the filter bar.

5. **See which filters you applied or remove filters**: The filters you select display in the filter bar. To clear filters, select the **X** next to the filter you want to clear.

6. **Change the time window**: Select a new time period from the time picker dropdown.

7. **View information for one specific app version**: Select the version that you want to see charts and lists for in the versions dropdown. By default, all available versions are shown.

The sections below describe what you can do on the request errors page.

> ## Triage errors tab [#triage-errors]

Reviewer: ```suggestion
## Triage tab [#triage-errors]
```

The triage tab shows an overview of unresolved errors and how they correlate with the error rate. The error group table lists groups of errors by occurrence and allows you to asssign them to users who can investigate and fix them. 

> ## Group errors tab [#group-errors]

Reviewer: ```suggestion
## Group request errors tab [#group-errors]
```

* **Identify patterns**: Examine the request error table for trends related to request domains, request paths, and frequency of occurrence.

* **Investigate individual request errors**: Select a request error report to view its distributed trace, event trail, response body, attributes, and other relevant details necessary for effective debugging.

* **Update error status**: Mark the request error as **Resolved** or assign it to a specific team member who can own and examine the issue, and then deploy a fix.

## Request error details [#error-details]

On the request errors page, click into a specific error to see:
* **User journeys**: Displays the different paths and actions a user performed that led to the error.
* **All occurrences chart**: Displays the frequency of a request error over the selected period, starting from its initial detection. You can view either the aggregated data of all occurrences or a breakdown by app version.
* **Error type breakdown**: Displays the distribution of this request errors across different operating system versions or affected devices.

> Resolved errors include a banner containing details on the user who resolved the request error and the resolution timestamp. Note that mobile monitoring's data retention policies apply, allowing you to filter by resolved errors for historical analysis when needed.

Author: ```suggestion
Resolved errors include a banner containing details on the user who resolved the request error and the resolution timestamp. Note that [mobile monitoring data retention policies](https://docs.newrelic.com/docs/data-apis/manage-data/manage-data-retention/) apply, allowing you to filter by resolved errors for historical analysis when needed.
```

<img
    title="Request errors details page"
    alt="Screenshot of the request errors details page"
    src="/images/errors-inbox_screenshot-crop_request-errors-details-page.webp"
/>

<figcaption>
    **<DNT>[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Mobile > Request errors</DNT>**: Select a request error to open the request errors details page.
</figcaption>

### Profiles [#error-profiles]

> Error profiles provide visual details about significant differences in the frequency of different values for HTTP error events. For each attribute, the error profile includes:

Reviewer: @ally-sassman where is this in the UI?

> Error profiles provide visual details about significant differences in the frequency of different values for HTTP error events. For each attribute, the error profile includes:

Author: ```suggestion
When looking at a specific error, the profiles section provides visual details about significant differences in the frequency of different values for HTTP error events. For each attribute, the error profile includes:
```

> When looking at a specific error, the profiles section provides visual details about significant differences in the frequency of different values for HTTP error events. For each attribute, the error profile includes:

Author: ```suggestion
When you look at a specific error,, the profiles section provides visual details about significant differences in the frequency of different values for HTTP error events. For each attribute, the error profile includes:
```

* A heatmap showing how the error's attribute is distributed for values that deviate the most
* A label comparing the error attribute's distribution to that of other errors

<img
    title="Request errors profiles"
    style={{align: "left" }}
    width="50%"
    alt="Screenshot of the request errors profiles section"
    src="/images/errors-inbox_screenshot-crop_request-errors-profiles.webp"
/>

<figcaption>

>     **<DNT>[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Mobile > Request errors</DNT>**: Select a request error to open the request errors details page and view the heatmap distribution of the attributes.

Reviewer: @ally-sassman this caption doesn't quite get me to the profiles section, does it?

>     **<DNT>[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Mobile > Request errors</DNT>**: Select a request error to open the request errors details page and view the heatmap distribution of the attributes.

Author: ```suggestion
    **<DNT>[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Mobile > Request errors</DNT> > (Select a specific error)**: Select a request error to open the request errors details page and view the heatmap distribution of the attributes.
```

</figcaption>

### Triage [#triage-errors]

> The triage section associates the specific error occurrence you are viewing with its [system-created error group](/docs/errors-inbox/errors-inbox/#groups). These system-created error groups are identified by a unique fingerprint. It is this unique fingerprint that allows you to triage error groups by status updates or assignments.

Reviewer: @ally-sassman where is this in the UI?

Author: These are all subsections of the request errors details page, so you'd click into an error or error group and see all of these sections on the popover (triage, profiles, heatmap, etc.)

> The triage section associates the specific error occurrence you are viewing with its [system-created error group](/docs/errors-inbox/errors-inbox/#groups). These system-created error groups are identified by a unique fingerprint. It is this unique fingerprint that allows you to triage error groups by status updates or assignments.

Author: ```suggestion
When looking at a specific error, the triage section associates the specific error occurrence you are viewing with its [system-created error group](/docs/errors-inbox/errors-inbox/#groups). These system-created error groups are identified by a unique fingerprint. It is this unique fingerprint that allows you to triage error groups by status updates or assignments.
```

> When looking at a specific error, the triage section associates the specific error occurrence you are viewing with its [system-created error group](/docs/errors-inbox/errors-inbox/#groups). These system-created error groups are identified by a unique fingerprint. It is this unique fingerprint that allows you to triage error groups by status updates or assignments.

Author: ```suggestion
When you look at a specific error, the triage section associates the specific error occurrence you are viewing with its [system-created error group](/docs/errors-inbox/errors-inbox/#groups). These system-created error groups are identified by a unique fingerprint. It is this unique fingerprint that allows you to triage error groups by status updates or assignments.
```

For more info on how error groups are created, see [How error groups work](/docs/errors-inbox/errors-inbox/#how-groups-work), and to learn more about status and assignments, see [Error tracking](/docs/errors-inbox/errors-inbox/#assign).

<img
    title="Request errors triage page"
    alt="Screenshot of the request errors triage page"
    src="/images/errors-inbox_screenshot-crop_request-errors-triage-page.webp"
/>

<figcaption>

>     **<DNT>[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Mobile > Request errors</DNT> > (Select a request error)**: From the request errors details page you can triage specific error occurrences by adding status updates or assignments.

Reviewer: @ally-sassman does this path take me to the Triage page?

</figcaption>

### Distributed trace [#distributed-trace]

> The distributed trace section tracks and observes your requests as they flow through your application, traversing various services to reach completion. By visualizing the entire request path across different services, you can rapidly identify failures or performance bottlenecks.

Reviewer: @ally-sassman where is this in the UI? Is everything on the main page?

> The distributed trace section tracks and observes your requests as they flow through your application, traversing various services to reach completion. By visualizing the entire request path across different services, you can rapidly identify failures or performance bottlenecks.

Author: ```suggestion
When looking at a specific error, the distributed trace section tracks and observes your requests as they flow through your application, traversing various services to reach completion. By visualizing the entire request path across different services, you can rapidly identify failures or performance bottlenecks.
```

> When looking at a specific error, the distributed trace section tracks and observes your requests as they flow through your application, traversing various services to reach completion. By visualizing the entire request path across different services, you can rapidly identify failures or performance bottlenecks.

Author: ```suggestion
When you look at a specific error, the distributed trace section tracks and observes your requests as they flow through your application, traversing various services to reach completion. By visualizing the entire request path across different services, you can rapidly identify failures or performance bottlenecks.
```

To learn more about how distributed tracing works, see [Track requests across your microservices](/docs/distributed-tracing/concepts/introduction-distributed-tracing/).

<img
    title="Request errors distribute trace section"
    alt="Screenshot of the request errors distribute trace section"
    src="/images/errors-inbox_screenshot-crop_request-errors-distributed-trace.webp"
/>

<figcaption>

>     **<DNT>[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Mobile > Request errors</DNT> > (Select a request error)**: From the request errors details page you view the distributed trace associated with that request.

Reviewer: @ally-sassman are distributed traces embedded in the main page?

</figcaption>

### Response [#response]

> A typical response to a request comprises a response header and response body, which together convey information about the success or failure of the request. The response header contains metadata about the server, while the response body holds information about the output, including a success or failure code and corresponding message.

Reviewer: @ally-sassman how are "Responses" different from distributed traces?

We capture the response body for requests whenever available, displaying it on the request errors details page to expedite debugging.

<img
    title="Request errors response section"
    alt="Screenshot of the request errors response"
    src="/images/errors-inbox_screenshot-crop_request-errors-response.webp"
/>

<figcaption>
    **<DNT>[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Mobile > Request errors</DNT> > (Select a request error)**: From the request errors details page you view the response body associated with that request.
</figcaption>

### Event trail [#event-trail]
 
The event trail provides a chronological log of all events leading up to the request error, which helps with root cause analysis. These can be events New Relic monitors by default, or custom events. The event trail is sorted chronologically, beginning with the oldest event, which is typically the app launch, but you can modify the following:

* Sort: Toggle between ascending and descending order

> * Event filtering: Filter by event type, like app launch, request, request error,  or user actions

Reviewer: ```suggestion
* Event filtering: Filter by event type, like `app launch`, `request`, `request error`,  or `user actions`.
```

> The event trail provides a chronological log of all events leading up to the request error, which helps with root cause analysis. These can be events New Relic monitors by default, or custom events. The event trail is sorted chronologically, beginning with the oldest event, which is typically the app launch, but you can modify the following:

Author: ```suggestion
When you look at a specific error, the event trail provides a chronological log of all events leading up to the request error, which helps with root cause analysis. These can be events New Relic monitors by default, or custom events. The event trail is sorted chronologically, beginning with the oldest event, which is typically the app launch, but you can modify the following:
```

> * Event details: Expand individual events to inspect their attributes, like errorType, responseTime, or requestUrl for request events.

Reviewer: ```suggestion
* Event details: Expand individual events to inspect their attributes, like `errorType`, `responseTime`, or `requestUrl` for request events.
```

After you've sorted and filtered your events, you can dig a little deeper into the events that lead up to the error by examining:

> * **Custom breadcrumbs**: Utilize the [Record breadcrumb SDK](/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/record-breadcrumb/) to create custom MobileBreadcrumb events. This allows you to log specific application interactions that may be relevant to investigating your error.

Reviewer: ```suggestion
* **Custom breadcrumbs**: Utilize the [Record breadcrumb SDK](/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/record-breadcrumb/) to create custom `MobileBreadcrumb` events. This allows you to log specific application interactions that may be relevant to investigating your error.
```

* **Handled exceptions**: Use the [Record Handled Exception SDK](/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/record-handled-exceptions/) methods for iOS and Android to annotate where exceptions are handled in your application. These annotations will automatically populate the crash event trail.

For guidance on enhancing crash event trails with custom data, see [Record breadcrumbs](/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/record-breadcrumb/).

To fully leverage our crash analysis tools, make sure to:

> 1. Use the mobile SDK to create custom MobileBreadcrumb or MobileHandledException events.

Reviewer: ```suggestion
1. Use the mobile SDK to create custom `MobileBreadcrumb` or `MobileHandledException` events.
```

2. Enable `MobileRequest` events for capturing network request data.

<img
    title="Request errors event trail"
    alt="Screenshot of the request errors event trail section"
    src="/images/errors-inbox_screenshot-crop_request-errors-event-trail.webp"
/>

<figcaption>
    **<DNT>[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Mobile > Request errors</DNT> > (Select a request error)**: From the request errors details page you can dig deeper into the events that led up to a specific error using our event trail.
</figcaption>

### Attributes [#attributes]

> Each request error sample provides detailed information about the request, response as well as the parameters that led to the request erroring.

Reviewer: @ally-sassman there's no mention of attributes here--only in the figcaption.

> Each request error sample provides detailed information about the request, response as well as the parameters that led to the request erroring.

Author: ```suggestion
Each request error sample includes a comprehensive set of attributes that provide detailed information about the request, response, and the specific parameters that triggered the error. These attributes offer valuable insights into the error's context and help in understanding the root cause.
```

> Each request error sample includes a comprehensive set of attributes that provide detailed information about the request, response, and the specific parameters that triggered the error. These attributes offer valuable insights into the error's context and help in understanding the root cause.

Author: ```suggestion
When you look at a specific error, each request error sample includes a comprehensive set of attributes that provide detailed information about the request, response, and the specific parameters that triggered the error. These attributes offer valuable insights into the error's context and help in understanding the root cause.
```
